packages/rspack/src/Compiler.ts

```diff
@@ -425,10 +425,19 @@ class Compiler {
 	/**
 	 * @param callback - signals when the call finishes
 	 */
-	run(callback: liteTapable.Callback<Error, Stats>) {
+	run(
+		callback: liteTapable.Callback<Error, Stats>,
+		options: {
+			modifiedFiles?: ReadonlySet<string>;
+			removedFiles?: ReadonlySet<string>;
+		} = {}
+	) {
 		if (this.running) {
 			return callback(new ConcurrentCompilationError());
 		}
+
+		this.modifiedFiles = options.modifiedFiles;
```

Comment:
Should we assign `undefined` to `this.modifiedFiles`? 

```suggestion
		if (options.modifiedFiles) {
				this.modifiedFiles = options.modifiedFiles;
		}
```